File: Eunleelee/2024-02/2월 19일/숫자의 표현.py
def solution(n):
    answer = 1

    for i in range(1, int(n/2)+1):
        a =0
        j = i
        while 1:
            a += j
            if a == n:
                answer += 1
                break
            if a > n:
                break
            j += 1

    return answer

File: jja6312/2월19일/숫자의표현.java
class Solution {
    static int answer = 0;

    public int solution(int n) {

        for (int i = 1; i <= n; i++) {
            repeatPlusToN(i, n);
        }

        return answer;
    }

    private void repeatPlusToN(int i, int max) {
        int sum = 0;
        int index = i;
        while (sum < max) {
            sum += index;
            index++;
        }

        if (sum == max) {
            answer++;
        }
    }
}